### PR TITLE
Adds a DM Quirks/Tricks section to Contributing.dm

### PR DESCRIPTION
**Pull request by RemieRichards**

### CODE REVIEW

**Comment by okaw** [comment]
quirky

**Comment by Thunder12345** [comment]
tricky

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -208,6 +208,48 @@ This prevents nesting levels from getting deeper then they need to be.
 
 Math operators like +, -, /, *, etc are up in the air, just choose which version looks more readable.
 
+###Dream Maker Quirks/Tricks:
+Like all languages, Dream Maker has its quirks, some of them are beneficial to us, like these
+
+* In-To for loops: ```for(var/i = 1, i <= some_value, i++)``` is a fairly standard way to write an incremental for loop in most languages (especially those in the C family) however DM's ```for(var/i in 1 to some_value)``` syntax is oddly faster than its implementation of the former syntax; where possible it's advised to use DM's syntax. (Note, the ```to``` keyword is inclusive, so it automatically defaults to replacing ```<=```, if you want ```<``` then you should write it as ```1 to some_value-1```).
+
+* Istypeless for loops: a name for a differing syntax for writing for-each style loops in DM, however it is NOT DM's standard syntax hence why this is considered a quirk. Take a look at this:
+```
```
Comment (JJRcop) [inline CONTRIBUTING.md]:
```DM
var/list/bag_of_items = list(sword, apple, coinpouch, sword, sword)
var/obj/item/sword/best_sword = null
for(var/obj/item/sword/S in bag_of_items)
	if(!best_sword || S.damage > best_sword.damage)
    		best_sword = S
```

\`\`\`DM

Comment (RemieRichards) [inline CONTRIBUTING.md]: This entire document is written as either:
raw `````````
or 
``````C++```
so...

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -208,6 +208,48 @@ This prevents nesting levels from getting deeper then they need to be.
 
 Math operators like +, -, /, *, etc are up in the air, just choose which version looks more readable.
```
Comment (swankcookie) [inline CONTRIBUTING.md]: Isn't multiplication  preferred over division  for speed?

Comment (RemieRichards) [inline CONTRIBUTING.md]: this is mentioned in the "Other Notes" section already, and isn't really a DM-specific quirk.
(also multiplication over division is only preferred when the operation terminates in decimal space (the resulting number is not recurring))

**Comment by tkdrg** [comment]
i swear about 50% of the branches in master are remie-made

**Pull request merged**